File: WorldFeed/Server/src/Services/Upload/WorldFeed.Upload.API/Application/Requesters/ITwitterRequest.cs
namespace WorldFeed.Upload.Application.Requesters
{
    using WorldFeed.Upload.API.Web;
    using WorldFeed.Upload.Application.Query;
    using WorldFeed.Upload.Client;

    public interface ITwitterRequest
    {
        ITwitterQuery Query { get; set; }

        ITwitterClientHandler TwitterClientHandler { get; set; }

        ITwitterExecutionContext ExecutionContext { get; set; }
    }
}

File: Sharebook/Client/src/app/core/Public/Models/Interfaces/ITwitterRequest.ts
import {ITwitterQuery} from "./ITwitterQuery";
import {ITwitterClientHandler} from "../../../Core/Web/ITwitterClientHandler";
import {ITwitterExecutionContext} from "../../../Core/Client/TwitterExecutionContext";

export interface ITwitterRequest {
  query: ITwitterQuery;
  twitterClientHandler: ITwitterClientHandler;
  executionContext: ITwitterExecutionContext;
}
